release/models/flex-algo/openconfig-flexalgo.yang

```diff
@@ -0,0 +1,397 @@
+      type flex-algo-metric-type;
+      default "IGP";
+      description
+        "Indicates which is the corresponding metric type for
+         this algorithm";
+    }
+    leaf calc-type {
+      type flex-algo-calc-type;
+      description
+        "Indicates the calculation type, if advertised";
+    }
+    leaf priority {
+      type uint8;
+      default "237";
+      description
+        "Indicates the priority for this algorithm";
+    }
+
+    leaf-list exclude {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups to exclude in path calculation.";
+    }
+    leaf-list include-all {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, all of which must be included";
+    }
+    leaf-list include-any {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, one of which must be included";
+    }
+    leaf-list srlg-names {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/srlgs/srlg/name";
+      }
+      description
+        "list of references to named SRLGs to be excluded";
+    }
+    leaf flags-tlv {
+      type boolean;
+      description
+        "Advertise flex-algo definition flags";
+    }
+  }
+
+  grouping flex-algo-definition-top {
+    description
+      "Top level grouping for flexible Algorithms and their definitions";
+    list algorithm {
+      key "name";
+      description
+        "List of Flexible Algorithms";
+      leaf name {
+        type leafref {
+          path "../config/name";
+        }
+        description
+          "Name for a Flexible Algorithm";
+      }
+      container config {
+        description
+          "Attributes for a Flexible Algorithm";
+        uses algorithm-attributes;
+      }
+      container state {
+        config false;
+        description
+          "Attributes for a Flexible Algorithm";
+        uses algorithm-attributes;
+      }
+    }
+  }
+
+  grouping flex-algo-interface-attributes-config {
+    description
+      "global level definitions for interfaces
+      on which TE is run";
+
+    leaf interface-id {
+      type oc-if:interface-id;
+      description
+        "Id of the interface";
+    }
+
+    leaf te-metric {
+      type uint32;
+      description
+        "TE specific metric for the link";
+    }
+
+    leaf-list srlg-membership {
+      type leafref {
+          path "../../../../flex-algo-global-attributes/srlgs/srlg/name";
+      }
+      description
+        "list of references to named shared risk link groups that the
+        interface belongs to.";
+    }
```

Comment:
you need delay metric as well.

Reply:
That is true. Few implementation allow for dynamic computation of min-delay and few allows a static configuration. I am going to skip this for now and add it in next iteration if that is ok?